### PR TITLE
Check if PV directory to be migrated exists on the source cluster

### LINKED ISSUE #90: Error when folder does not exist to be transferred
Below is from an issue Keith Tenzer saw:

`[{"name": "proxydata-proxysql-1", "namespace": "cisapihub-appsrvc-qa", "stdout": "opening connection using: ssh -p 2222 -o StrictHostKeyChecking=no -i \"~/.ssh/dest_key\" -l root aa650e42f13034e3aa8289a61f3316cd-688183305.us-east-1.elb.amazonaws.com rsync --server -vvlHogDtpre.iLsfxC --partial . /mnt/cisapihub-appsrvc-qa/proxydata-proxysql-1  (16 args)", "stderr": "ssh_exchange_identification: Connection closed by remote host\r\nrsync: connection unexpectedly closed (0 bytes received so far) [sender]\nrsync error: unexplained error (code 255) at io.c(226) [sender=3.1.2]", "rc": "255"}]`

> This was caused by a folder not existing. We worked around issue but in pvc script would be good to check if the folder exists before rsyncing  or maybe even the pod where pvc is mounted is in running state and throwing a more meaningful error or potentially trapping the rsync error instead of the ssh error, connection reset.

### PR DESCRIPTION
**Pull request by pranavgaikwad**
- adds new check to ensure directory exists on the source cluster 
- fixes success/failure result collection logic for cases where Rsync step is never reached

Fixes #90

### CODE REVIEW

File: 3_run_rsync/tasks/rsync.yml [hunk 1]
```diff
@@ -46,9 +46,19 @@
...
+    stat:
+      path: "{{ mig_source_data_location }}"
+    register: source_dir_stat
+
+  - when: not source_dir_stat.stat.exists
+    fail:
+      msg: |
+        The directory {{ mig_source_data_location }} does not exist on the source cluster.
+        Skipping to the next PV...
+
```
Comment (alaypatel07) [inline 3_run_rsync/tasks/rsync.yml:59]: @pranavgaikwad wonder if we should add this PVs to failed PV list so users can revisit in the next run

Comment (pranavgaikwad) [reply 3_run_rsync/tasks/rsync.yml:59]: @alaypatel07 `fail` signal already takes care of that. Check out the new conditional logic on line 81 Basically, any failure inside this `block` triggers the failed collection task

Comment (alaypatel07) [reply 3_run_rsync/tasks/rsync.yml:59]: okay makes sense, thanks

**Comment by alaypatel07** [comment]
LGTM

**Review by alaypatel07 — APPROVED** [review]

**Pull request merged**